language/compiler/bytecode_source_map/src/disassembler.rs

```diff
@@ -111,14 +121,96 @@ impl<Location: Clone + Eq + Default> Disassembler<Location> {
                 .get(ty_param_index as usize)
                 .ok_or_else(|| {
                     format_err!(
-                        "Type parameter index out of bounds while disassembling type signature"
+                        "Type parameter index {} out of bounds while disassembling type signature",
+                        ty_param_index
                     )
                 })?
                 .0
                 .to_string(),
         })
     }
 
+    fn name_for_field(&self, field_idx: FieldDefinitionIndex) -> Result<(String, String)> {
+        let field_def = self.source_mapper.bytecode.field_def_at(field_idx);
+        let struct_def_idx = self
+            .source_mapper
+            .bytecode
+            .struct_defs()
+            .iter()
+            .position(|struct_def| struct_def.struct_handle == field_def.struct_)
+            .ok_or_else(|| format_err!("Unable to find struct for MutBorrowField"))?;
+        let name = self
+            .source_mapper
+            .bytecode
+            .identifier_at(field_def.name)
+            .to_string();
+        let struct_source_info = self
+            .source_mapper
+            .source_map
+            .get_struct_source_map(StructDefinitionIndex(struct_def_idx as TableIndex))?;
+        let field_type_sig = self
+            .source_mapper
+            .bytecode
+            .type_signature_at(field_def.signature);
+        let ty = self.disassemble_sig_tok(
+            field_type_sig.0.clone(),
+            &struct_source_info.type_parameters,
+        )?;
+        Ok((name, ty))
+    }
+
+    fn class_ty_info(
+        &self,
+        class_idx: &StructDefinitionIndex,
+        types_idx: &LocalsSignatureIndex,
+    ) -> Result<String> {
+        let struct_definition = self.get_struct_def(*class_idx)?;
+        let struct_source_map = self
+            .source_mapper
+            .source_map
+            .get_struct_source_map(*class_idx)?;
+        let locals_signature = self.source_mapper.bytecode.locals_signature_at(*types_idx);
+        let type_arguments = locals_signature
+            .0
+            .iter()
+            .map(|sig_tok| {
+                Ok(self.disassemble_sig_tok(sig_tok.clone(), &struct_source_map.type_parameters)?)
+            })
+            .collect::<Result<Vec<String>>>()?;
+
+        let struct_handle = self
+            .source_mapper
+            .bytecode
+            .struct_handle_at(struct_definition.struct_handle);
+        let name = self
+            .source_mapper
+            .bytecode
+            .identifier_at(struct_handle.name)
+            .to_string();
+        Ok(format!(
+            "{idx}, {name}{ty_instantiation}",
+            idx = class_idx,
+            name = name,
+            ty_instantiation = Self::enclose_ty_params(type_arguments)
+        ))
+    }
+
+    fn enclose_ty_params(ty_params: Vec<String>) -> String {
```

Comment:
Might be nitpicking, but is it necessary for `enclose_ty_params` and `format_ret_type` to take ownership of the vector?

Reply:
Not strictly necessary. I'll update.